openedx/core/djangoapps/schedules/management/commands/__init__.py

```diff
@@ -27,10 +31,18 @@ def add_arguments(self, parser):
             help='Send all emails to this address instead of the actual recipient'
         )
         parser.add_argument('site_domain_name')
+        parser.add_argument(
+            '--duration',
+            type=int,
+            help='The duration for which weekly emails are sent (days).',
+        )
 
     def handle(self, *args, **options):
         self.log_debug('Args = %r', options)
 
+        if 'duration' in options.keys():
+            self.offsets = range(-7, -options['duration'], -7)
```

Comment:
You shouldn't need `.keys()` here. Just `if 'duration' in options:` should be sufficient.